### DISCUSSION TITLE
ST7789 & NodeMCU esp8266

### DISCUSSION THREAD

**Discussion by minimoibientot** [Bodmer/TFT_eSPI#2359] [Compatible support libraries]
good morning

I bought this card : NodeMcu v3 CH340 ESP8266
https://fr.aliexpress.com/item/1005004527993328.html?spm=a2g0o.order_list.order_list_main.17.3aa9caa4NUZJts&gatewayAdapt=glo2fra
and this screen : 1.44' 128*128 RGB TFT ST7789VW
https://fr.aliexpress.com/item/32996979276.html?spm=a2g0o.order_list.order_list_main.29.3aa95e5b7YnT9t&gatewayAdapt=glo2fra

Connected PINs :
GND => G
VCC => 3v
SCL => D5
SDA => D7
RES => D2
DC => D1
CS => NC
BLK => NC

user_setup_select.h
#define ST7789_2_DRIVER
#define TFT_WIDTH 128
#define TFT_HEIGHT 128
#define TFT__DC 5
#define TFT__RST 4
![20230127_163141](https://user-images.githubusercontent.com/123747460/215126019-28ff1423-c5a0-4ddc-a3fd-a492e52b9c6a.jpg)

I find the colors poorly defined
and and dark luminosity

I have to connect or configure otherwise?

thank you

**Comment by Bodmer** [comment]
The display brightness is dependant on the backlight LED brightness and is not controlled by the processor. It is dependant on the display design and supply voltage.

The image does show inverted collours. You need to add ONE of the following lines to the setup file to corect this:

#define TFT_INVERSION_ON
#define TFT_INVERSION_OFF

**Reply by minimoibientot** [reply]
thank you
ON :
![20230130_101219](https://user-images.githubusercontent.com/123747460/215436537-e1959265-97b2-4450-9f4f-363a791094cc.jpg)
OFF :
![20230130_101239](https://user-images.githubusercontent.com/123747460/215436568-9a788f53-3917-4a2b-b7fa-4b0b63c6e3a9.jpg)

plugging in CS or BLK can't change anything?

Connected PINs is correct?
GND => G
VCC => 3v
SCL => D5
SDA => D7
RES => D2
DC => D1
CS => NC
BLK => NC
